docs/developer/index.rst

```diff
@@ -3,7 +3,6 @@ Developer
 
```

Comment:
I'm not sure we need this file. We have https://github.com/python-pillow/Pillow/blob/master/CONTRIBUTING.md for general things a contributor should check, whether bugs or code. A good thing about that file is it's autolinked from the "Please review the guidelines for contributing to this repository." banner on https://github.com/python-pillow/Pillow/issues/new and when creating PRs. (The CI skip note can go in there.)

For releasers, we have https://github.com/python-pillow/Pillow/blob/master/RELEASING.md